netlogo-gui/src/test/agent/WorldTests.scala

```diff
@@ -24,6 +26,30 @@ class WorldTests extends FunSuite with AbstractTestWorld {
   override def makeLink(world: World, ends: Array[Int]) =
     new Link(world, world.getTurtle(ends(0)),
              world.getTurtle(ends(1)), world.links)
+
+  def makeBreededTurtle(world: World, breedName: String) =
+    new Turtle(world, world.getBreed(breedName), 0.0, 0.0)
+
+  def makeBreededLink(world: World, breedName: String, end1: Int, end2: Int) =
+    new Link(world, world.getTurtle(end1), world.getTurtle(end2),
+      world.getLinkBreed(breedName))
+
+  def makeWorld(dimensions: WorldDimensions, program: Program) = {
+    val w = new World() {
+      createPatches(dimensions)
+      realloc()
+    }
+    w.program(program)
+    w.realloc()
+    w
+  }
+
+  def changeProgram(w: World, program: Program) = {
+    w.rememberOldProgram()
```

Comment:
That's done by `CompilerManager` and it's actually called immediately following a successful compilation (so subsequent compilations don't have to worry about it). I'll see if I can figure out a way to test it tomorrow morning, but I'm a bit unsure whether I can get something to work since it's heavily entwined with the awt event thread.